src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DoNotUseEnumerableMethodsOnIndexableCollectionsInsteadUseTheCollectionDirectly.Fixer.cs

```diff
@@ -51,12 +52,17 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
             }
 
             var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
-            if (semanticModel.GetOperation(invocationNode, context.CancellationToken) is not IInvocationOperation invocationOperation)
+            var operation = semanticModel.GetOperation(invocationNode, context.CancellationToken);
+            if (!TryExtractInvocationOperation(operation, out var invocationOperation))
             {
                 return;
             }
 
             var collectionSyntax = invocationOperation.GetInstanceSyntax();
+
+            if (invocationOperation is { Instance: IConversionOperation { Operand: IConditionalAccessInstanceOperation conditionalAccessInstance } })
+                collectionSyntax = conditionalAccessInstance.Syntax;
+
```

Comment:
I don't know how to adequately explain, what is going on here. This is due to VB operation tree structure and is just required for it to work

Reply:
Definitely needs a comment in code.